packages/components/atoms/f-button/src/components/Button.vue

```diff
@@ -14,19 +14,33 @@
         v-bind="$attrs"
         :aria-live="getAriaLive"
         :aria-busy="isLoading"
-        v-on="!isLoading && $listeners"
-    >
+        v-on="!isLoading && $listeners">
         <span
             v-if="isLoading"
             :class="$style['c-spinner']"
             :data-test-id="`${componentType}-spinner`" />
-
         <span
-            :class="[
-                $style['o-btn-text'],
-                (isLoading ? $style['o-btn-text--hidden'] : '')
-            ]">
-            <slot />
+            :class="$style['o-button-content']">
+
+            <span
+                v-if="hasIcon && hasIcon === 'leading'"
+                :class="(hasIcon === 'leading' ? [$style['o-btn-icon'], $style['o-btn-icon--leading']] : '')">
+                <slot name='leading-icon' />
+            </span>
+
+            <span
+                :class="[
+                    $style['o-btn-text'],
+                    (isLoading ? $style['o-btn-text--hidden'] : '')
```

Comment:
Could the classes for these three spans be 
```
{ [$style['o-btn-icon'], $style['o-btn-icon--leading']] :   hasIcon === 'leading' }
{ [$style['o-btn-text--hidden']] : isLoading }
{ [$style['o-btn-icon'], $style['o-btn-icon--trailing']] : hasIcon === 'trailing' }
```

Reply:
I agree, I think they should be. Not sure where the ternary approach came in but we should be using the Vue syntax for conditional classes IMO.

Reply:
It didn't allow me to use two classes in one list so I had to split them back in two separate checks

Reply:
Oh right, I didn't catch that there were two classes. In that case, they should be broken out into two separate lines I reckon. Also, the `hasIcon === 'leading'` & `hasIcon === 'trailing'` expressions should be moved to computed props to keep logic out of the template.